config/config.go

```diff
@@ -432,32 +446,45 @@ func (s *SourceBlockchain) Validate(destinationBlockchainIDs *set.Set[string]) e
 	}
 	s.subnetID = subnetID
 
-	// Validate and store the allowed destinations for future use
-	s.supportedDestinations = set.Set[ids.ID]{}
-
 	// If the list of supported destinations is empty, populate with all of the configured destinations
 	if len(s.SupportedDestinations) == 0 {
 		for _, blockchainIDStr := range destinationBlockchainIDs.List() {
-			blockchainID, err := ids.FromString(blockchainIDStr)
-			if err != nil {
-				return fmt.Errorf("invalid blockchainID in configuration. error: %w", err)
-			}
-			s.supportedDestinations.Add(blockchainID)
+			s.SupportedDestinations = append(s.SupportedDestinations, &SupportedDestination{
```

Comment:
I think we should avoid cases of not initializing the full struct. How about a `NewSupportedDestination` func that takes in both fields and does the conversion for blockchainID, and sets the access variables.

Reply:
Up until the loop starting below this one, none of the `SupportedDestinations` are fully populated, so this check is equivalent to the user manually providing all of the destination chain IDs with no allowed addresses specified. Then in the following loop we populate all of the elements of the list.

Adding a `NewSupportedDestination` method call here doesn't make much sense IMO, since the `SupportedDestination` objects already exist.

Reply:
makes sense since the object already exists. My main gripe with this file is that the types `SourceBlockchinID` `DestinationBlockchainID` `ManualWarpMessage` `Config`, etc. and all their helper and validation functions all stuck into one file. Since the code has expanded more than we originally had, would want to distribute to different files. Feel free to defer to new issue though.

Reply:
Agreed that this file should be split up. I'll defer that to a later issue though: https://github.com/ava-labs/awm-relayer/issues/251